### PR TITLE
[FIX] web_editor: unneed size set when adding a row above a table

### PR DESCRIPTION
**Pull request by Jinjiu96**
Reproduction:

1. In project -> task, create a new task
2. In the description, make a table of 1 row 2 columns, type two line long string in the second cell
3. Create a row above, type anything short, save
4. Add the portal user as follower, e.g. search user joel
5. In an incognito tab log in with portal portal, check the task and the table is out of the field

Fix: Only set the width of the cells when it’s the first row and there’s other preset style of width for existing cells

task-3559104

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js [hunk 1]
```diff
@@ -720,8 +720,9 @@ export const editorCommands = {
         }));
         referenceRow[beforeOrAfter](newRow);
         newRow.style.height = referenceRowHeight + 'px';
-        // Preserve the width of the columns (applied only on the first row).
-        if (getRowIndex(newRow) === 0) {
+        // Preserve the width of the columns (applied only on the first row and when any cell has fixed width).
+        const withFixedWidthCell = [...cells].map(cell => !cell.style.width).some(el => el === false);
+        if (getRowIndex(newRow) === 0 && withFixedWidthCell) {
```
Comment (Goaman) [inline addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js:725]: I would remove the use of `referenceRowWidths`.
That logic seems flawed.
If there was no width on the columns, it means that the user has never resized the width of a column, we should not retrieve the `clientWidth` and force a particular width.

Without the use of `referenceRowWidths`, the code could be rewritten like so:

```js
if (getRowIndex(newRow) === 0) {
            let columnIndex = 0;
            for (const newColumn of newRow.children) {
                newColumn.style.width = cells[columnIndex].style.width;
                cells[columnIndex].style.width = '';
                columnIndex++;
            }
        }
```

@Zynton are you ok with those changes or is there a hidden reason behind using `clientWidth`?

Comment (Zynton) [reply addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js:725]: It's not clear and I can't find a trace of a discussion about it. I do remember there being a reason though but I guess we can try removing it and test extensively to make sure nothing breaks. If nothing breaks we're good...

Comment (Jinjiu96) [reply addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js:725]: Thanks guys! I will adapt the code, test it and put it back to TestIN so we make sure not to break anything :pray:

Comment (Jinjiu96) [reply addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js:725]: @Goaman I found another related issue that, the table is shown improperly after we change the width of a column (spill out the form)


I will try to find a solution for this and push it to this PR

**Comment by Zynton** [comment]
@robodoo r+

**Pull request closed without merging**